core/src/main/scala/za/co/absa/spline/harvester/builder/write/WriteCommandExtractor.scala

```diff
@@ -74,9 +79,18 @@ class WriteCommandExtractor(pathQualifier: PathQualifier, session: SparkSession)
 
       case cmd: InsertIntoHadoopFsRelationCommand =>
         val path = cmd.outputPath.toString
-        val format = cmd.fileFormat.toString
         val qPath = pathQualifier.qualify(path)
-        WriteCommand(cmd.nodeName, SourceIdentifier(Some(format), qPath), cmd.mode, cmd.query, cmd.options)
+        val fileFormat = cmd.fileFormat
+        fileFormat match {
+          case `_: SparkAvroSourceExtractor`(avro) =>
```

Comment:
Naming convention:
This notation ``` `_: Something` ``` supposes to simulate Scala type matching pattern, that you would normally use if the type `Something` were guaranteed to be available in the runtime. In this case we don't use `Extractor` or any other suffix to the type name, to make it as close to the native Scala syntax as possible. This is the only case we use that "weird" notation. So either remove the suffix or the back ticks.

Reply:
Otherwise LGTM. Thanks.